### ISSUE TITLE
Update date should actually be manual

### ISSUE THREAD

**Issue by pwalsh** [okfn/ibp-explorer#74]
# Client requests
- the “data as of” date should be April 30, 2016. This date should be static until we post new data (which wouldn’t be until 2017). This is different from the Tracker pilot website, which had that date automatically ticking forward a month at a set time every month.
# We need
- To understand what it means by "new data is posted" - via API, via?
- @dumyan to confirm if this is something we can set manually only when new data is posted.

**Comment by pwalsh** [comment]
# client

“Posting of new data” means updating the document availability/tracker tab with the updated Section 1 of the OBS for all the countries/ We are going to do so three more times until the end of 2017. Maybe I need to better understand how the API works: if it picks up data that have reached a certain stage in the research process (for example, now we could only pull the data once it reached goal n.5), then the API will “know”. But the moment in which the API pulls the data does not correspond to the actual “update” date. This is because there is a lag. The API now pulled data on September 2X, but the data only reflect what happened as of April 30, 2016.

# actions

- @pwalsh I don't follow, but seeing as this "posting" is related to the exposure of new data via the API (as I understand it), we will need some way to "know" that such a "posting" has occurred. I'm also unclear on the "lag" between April 30 and September 2x.

- @dumyan question: can you help me make sense of this?

**Comment by dumyan** [comment]
It seems to me that one part of the problem is that we don't have information about the API behavior to directly proceed from here. We ought to know what are the Sections mentioned and how the API will correspond to them.

So, if the API updates when the "updating cycle" is completed in 2017 (the sections mentioned) we setup the tracker to check the API every day and compare to see if there are any new (or modified?) objects and properties are present and update the content and changing the date.

If the API updates when a section is updated we can try to logically separate section updates from complete updates (all sections updated), but for this we will need better understanding of the updating process of the API itself to decouple the new data from the old and verify its completeness.

The most straight forward option is knowing when the data is updated trough the API from some endpoint and use that information to know whether we should update the site.

As for the current date that is on the tracker, it is simply the last date when the site has updated its cache from the API because I don't think we have any way of knowing the exact date the update has occurred (because of the 6 months refreshing period, we could catch the exact date if we check the API once a day for example, but this doesn't make any sense since the new update should happen in 2017).

Manually changing the date is of course possible, but I think it's cumbersome without implementing some authorization mechanism so a user can log in and set the date. The other option I have on my mind is hardcoding the dates and updating it every update.

**Comment by pwalsh** [comment]
Was the old code a manual update? I think that we may have to do a manual update.

**Comment by dumyan** [comment]
The Aquarium API was always returning the latest changes made int he database. The tracker was updating its cache every 1 hour but it implements [logic to modify the displayed date](https://github.com/keitaroinc/open-budget-survey-tracker/blob/2015/routes/index.js#L225).

It seems to me that they pushed the changes internally on the 22 each month, so that is how the date was in sync every month.

**Comment by pwalsh** [comment]
So, the april date that shows on the current live site, came from the aquarium API?

**Comment by dumyan** [comment]
I think we cleared this question elsewhere, but the answer is that the date is actually when the tracker updated its cache files from the API with some logic which modifies the date according to what day in the month currently is.

We should note that now according to the last email from David that now we will need to manually adjust the date and we will be notified when to do this. 

I have to mention that this means if the API is updated before we change the date **and** either the cache is missing or the date of creation of the cache file present is older then 6 months (we have changed this period from 1 to 6 months) there can be a time when the content will get updated but the older date will be available on the site.

To remedy this, I think we can easily save the current data on S3 and fetch it from there until we are notified for an update, and then also store it there.
